src/main/java/frc/utils/ToleranceCalculations.java

```diff
@@ -0,0 +1,17 @@
+package frc.utils;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.constants.MathConstants;
+
+public class ToleranceCalculations {
+
+	public static boolean isRotation2dInTolerance(Rotation2d tolerance, Rotation2d rotation, Rotation2d wantedRotation) {
+		return Math.abs(wantedRotation.getRadians() - rotation.getRadians()) <= tolerance.getRadians()
+			|| Math.abs(wantedRotation.getRadians() - rotation.getRadians()) >= MathConstants.FULL_CIRCLE.getRadians() - tolerance.getRadians();
+	}
+
+	public static boolean isDoubleInRange(double tolerance, double value, double maximumValue, double minimumValue) {
```

Comment:
So why tolerance?
Also see MathUtil.isNear()

Reply:
In range but with tolernce in case i want to change it.

Reply:
also isNear is for being around a value, i want being inside a range of values